_ingest-pipelines/processors/urldecode.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: default
+title: URL decode
+parent: Ingest processors
+nav_order: 320
+---
+
+# URL decode processor
+
+The `urldecode` processor is useful for decoding URL-encoded strings that may be present in log data or other text fields. This can make the data more readable and easier to analyze, especially when dealing with URLs or query parameters that contain special characters or spaces.
+
+The following is the syntax for the `urldecode` processor:
+
+```json
+{
+  "urldecode": {
+    "field": "field_to_decode",
+    "target_field": "decoded_field"
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `urldecode` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field`  | Required  | The field containing the URL-encoded string to be decoded. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
```

Comment:
@gaobinlong Are template snippets applicable for this processor's parameters?

Reply:
No, not applicable, both the `file` and `target_field` do not support template snippets.